src/lib/util/view.h

```diff
@@ -0,0 +1,154 @@
+
+		typedef std::random_access_iterator_tag iterator_category;
+
+		iterator(T *ptr) : m_ptr(ptr) { }
+		iterator(const iterator &that) : m_ptr(that) { }
+
+		iterator& operator=(const iterator &that) { m_ptr = that.m_ptr; return *this; }
+		bool operator==(const iterator &that) const { return m_ptr == that.m_ptr; }
+		bool operator!=(const iterator &that) const { return m_ptr != that.m_ptr; }
+		bool operator<(const iterator &that) const { return m_ptr < that.m_ptr; }
+		bool operator>(const iterator &that) const { return m_ptr > that.m_ptr; }
+		bool operator<=(const iterator &that) const { return m_ptr <= that.m_ptr; }
+		bool operator>=(const iterator &that) const { return m_ptr >= that.m_ptr; }
+		iterator operator+(size_type i) const { return iterator(m_ptr + i); }
+		iterator operator-(size_type i) const { return iterator(m_ptr - i); }
+		difference_type operator-(const iterator &that) const { return m_ptr - that.m_ptr; }
+
+		iterator& operator++() { m_ptr++; return *this; }
+		iterator operator++(int) { iterator result = *this; m_ptr++; return result; }
+
+		reference operator*() const { return *m_ptr; }
+		pointer operator->() const { return m_ptr; }
+		operator T*() const { return m_ptr; }
+
+	private:
+		T *m_ptr;
+	};
+
+	class const_iterator
+	{
+	public:
+		typedef std::ptrdiff_t difference_type;
+		typedef T value_type;
+		typedef const T &const_reference;
+		typedef const T *const_pointer;
+
+		typedef std::random_access_iterator_tag iterator_category;
+
+		const_iterator(const T *ptr) : m_ptr(ptr) { }
+		const_iterator(const iterator &that) : m_ptr(that) { }
+		const_iterator(const const_iterator &that) : m_ptr(that) { }
+
+		const_iterator& operator=(const const_iterator &that) { m_ptr = that.m_ptr; return *this; }
+		bool operator==(const const_iterator &that) const { return m_ptr == that.m_ptr; }
+		bool operator!=(const const_iterator &that) const { return m_ptr != that.m_ptr; }
+		bool operator<(const const_iterator &that) const { return m_ptr < that.m_ptr; }
+		bool operator>(const const_iterator &that) const { return m_ptr > that.m_ptr; }
+		bool operator<=(const const_iterator &that) const { return m_ptr <= that.m_ptr; }
+		bool operator>=(const const_iterator &that) const { return m_ptr >= that.m_ptr; }
+		const_iterator operator+(size_type i) const { return const_iterator(m_ptr + i); }
+		const_iterator operator-(size_type i) const { return const_iterator(m_ptr - i); }
+		difference_type operator-(const const_iterator &that) const { return m_ptr - that.m_ptr; }
+
+		const_iterator& operator++() { m_ptr++; return *this; }
+		const_iterator operator++(int) { const_iterator result = *this; m_ptr++; return result; }
+
+		const_reference operator*() const { return *m_ptr; }
+		const_pointer operator->() const { return m_ptr; }
+		operator T*() const { return m_ptr; }
+
+	private:
+		const T *m_ptr;
+	};
+
+	view(T *ptr, std::size_t size)
+		: m_begin(ptr)
+		, m_end(ptr + size)
+	{
+	}
+
+	view(const view &that)
```

Comment:
As with the iterators, there's no point doing this explicitly, just use `= default` to get the standard copy constructor.